### DISCUSSION TITLE
Multiple Relay Server WebSocket connection

### DISCUSSION THREAD

**Discussion by arvindk-dev** [practable/relay#55] [Q&A]
Hello,

Does this relay server allow to open multiple websocket connections between multiple experiments and client browsers.

Please let me know if the relay server supports below scenario.

**Our Scenario:**
We are trying to fit this relay server in a system where we need to connect and stream video from multiple cameras to multiple clients. Basically the relay server needs to open an unique websocket connection for each camera based on client request.

**Comment by timdrysdale** [comment]
That's exactly the scenario we have for remote labs, multiple unique camera streams and multiple clients in a many to many relationship.  However there is also a need for a management layer that knows the topic for each of the streams, and provides JWT tokens to the clients for the streams they want to access ([practable/book](HTTPS://GitHub.com/practable/book)). The streamers also need to be configured and provided with a JWT for correct topic - we do this via our experiment configuration procedures which are still somewhat manual (we have to build the hardware in each case so assigning a topic and JWT to the `host` manually are a small acceptable overhead for a long lived stream). If you have different use case you can still use relay but provide your own management layer.
